### ISSUE TITLE
Unhandled Exception when parsing completes after CustomTextSpanNotifier.disposed()

### ISSUE THREAD

**Issue by kaboc** [kaboc/flutter_custom_text#48]
1.4.2

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: A CustomTextSpanNotifier was used after being disposed.
Once you have called dispose() on a CustomTextSpanNotifier, it can no longer be used.
#0      ChangeNotifier.debugAssertNotDisposed.<anonymous closure> (package:flutter/src/foundation/change_notifier.dart:179:9)
#1      ChangeNotifier.debugAssertNotDisposed (package:flutter/src/foundation/change_notifier.dart:186:6)
#2      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:412:27)
#3      _ValueNotifier.updateValue (package:custom_text/src/span/text_span_notifier.dart:34:7)
#4      CustomTextSpanNotifier.buildSpan (package:custom_text/src/span/text_span_notifier.dart:80:5)
#5      _CustomTextState._parse (package:custom_text/src/text.dart:591:9)
<asynchronous suspension>
#6      _CustomTextState._buildSpans (package:custom_text/src/text.dart:512:7)
```

This happens probably when CustomText is removed during parsing. It appears older versions (<= 1.3.1) did not have the bug. It is easy to just suppress the exception, but it should be better to find the root cause.

**Comment by kaboc** [comment]
### Reproducible code

CustomText is removed before parsing ends and the exception is thrown.

```dart
import 'package:flutter/material.dart';

import 'package:custom_text/custom_text.dart';

void main() => runApp(const App());

class App extends StatefulWidget {
  const App();

  @override
  State<App> createState() => _AppState();
}

class _AppState extends State<App> {
  bool _visible = true;

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) {
      setState(() => _visible = false);
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: SafeArea(
          child: _visible
              ? CustomText(
                  'abcde',
                  parserOptions: ParserOptions.external(_parser),
                  definitions: const [
                    TextDefinition(matcher: PatternMatcher('')),
                  ],
                )
              : const SizedBox.shrink(),
        ),
      ),
    );
  }

  Future<List<TextElement>> _parser(String text) async {
    await Future<void>.delayed(const Duration(seconds: 1));
    return [TextElement(text)];
  }
}
```

**Comment by kaboc** [comment]
The check for notifier disposal was mistakenly removed by #45.

---

26691edb5929edb98dcf26c091234583bb3de4c9

https://github.com/kaboc/flutter_custom_text/blob/26691edb5929edb98dcf26c091234583bb3de4c9/lib/src/span/text_span_notifier.dart#L35-L40

---

d7d46f85a564584d430664c847e7f5fe3bfb3232 (#45)

https://github.com/kaboc/flutter_custom_text/blob/d7d46f85a564584d430664c847e7f5fe3bfb3232/lib/src/span/text_span_notifier.dart#L25-L29

https://github.com/kaboc/flutter_custom_text/blob/d7d46f85a564584d430664c847e7f5fe3bfb3232/lib/src/span/text_span_notifier.dart#L60-L64

The `_disposed` flag is only used here (as fix for #6):

https://github.com/kaboc/flutter_custom_text/blob/d7d46f85a564584d430664c847e7f5fe3bfb3232/lib/src/span/text_span_notifier.dart#L113-L123
